File: src/test/java/org/dedda/bratwurst/parse/ParserTest.java
package org.dedda.bratwurst.parse;

import org.dedda.bratwurst.BratwurtstTestcase;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * Created by dedda on 10/21/15.
 *
 * @author dedda
 */
public class ParserTest extends BratwurtstTestcase {

    @Test
    public void testInsertIntoArray() throws Exception {
        String[] array = new String[]{
                "==>",
                "|65|",
                "%include%",
                "|67|",
                "<=="
        };
        String[] toInsert = new String[]{
                "|1|",
                "|2|",
                "|3|"
        };
        String[] expected = new String[]{
                "==>",
                "|65|",
                "|1|",
                "|2|",
                "|3|",
                "|67|",
                "<=="
        };
        int index = 2;
        Parser parser = new Parser(null);
        String[] actual = parser.insertIntoArray(array, toInsert, index);
        assertEquals(expected.length, actual.length);
        for (int i = 0; i < actual.length; i++) {
            assertEquals(expected[i], actual[i]);
        }
    }
}

File: src/test/java/org/dedda/bratwurst/parse/ParserTest.kt
package org.dedda.bratwurst.parse

import org.dedda.bratwurst.BratwurtstTestcase
import org.testng.Assert
import org.testng.annotations.Test

/**
 * Created by dedda on 10/21/15.
 *
 * @author dedda
 */
class ParserTest : BratwurtstTestcase() {
    @Test
    @Throws(Exception::class)
    fun testInsertIntoArray() {
        val array = arrayOf(
                "==>",
                "|65|",
                "%include%",
                "|67|",
                "<=="
        )
        val toInsert = arrayOf(
                "|1|",
                "|2|",
                "|3|"
        )
        val expected = arrayOf(
                "==>",
                "|65|",
                "|1|",
                "|2|",
                "|3|",
                "|67|",
                "<=="
        )
        val index = 2
        val parser = Parser(null)
        val actual = parser.insertIntoArray(array, toInsert, index)
        Assert.assertEquals(expected.size, actual.size)
        for (i in actual.indices) {
            Assert.assertEquals(expected[i], actual[i])
        }
    }
}